### PR TITLE
[New Rule] Custom Gmail Route Created or Modified - Google Workspace

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/detection-rules/issues/2226

## Summary
Gmail is by default the mail routing service in Google Workspace. By default, all email from current Google Workspace users with accounts are routed through a domain's mail server for inbound and outbound mail. Adversaries can add a custom e-mail route for outbound mail to route these e-mails to their own inbox of choice for data gathering. This allows adversaries to capture sensitive information from e-mail and potential attachments, such as invoices or payment documents.

KQL query screenshot:

<details>

Screen Shot 2022-09-13 at 2 40 04 PM

</details>

Document example:

<details>

```
{
  "_index": ".ds-logs-google_workspace.admin-default-2022.08.24-000002",
  "_id": "voDTrbgSQv6EfDoOPtqZDM+tQuI=",
  "_version": 1,
  "_score": 0,
  "_source": {
    "agent": {
      "name": "ubuntu-server-tdejesus",
      "id": "f4011165-a1b8-4d8e-b902-56ba8835cc28",
      "type": "filebeat",
      "ephemeral_id": "328620d6-3e81-4b39-863a-bf064564dfa0",
      "version": "8.4.0"
    },
    "elastic_agent": {
      "id": "f4011165-a1b8-4d8e-b902-56ba8835cc28",
      "version": "8.4.0",
      "snapshot": false
    },
    "source": {
      "geo": {
        "continent_name": "North America",
        "region_iso_code": "US-OH",
        "city_name": "North Canton",
        "country_iso_code": "US",
        "country_name": "United States",
        "region_name": "Ohio",
        "location": {
          "lon": -81.3798,
          "lat": 40.7961
        }
      },
      "as": {
        "number": 10796,
        "organization": {
          "name": "TWC-10796-MIDWEST"
        }
      },
      "ip": "2603:6010:ed07:5cb7:73:3153:929f:34bf",
      "user": {
        "domain": "dejesus-gw-rule-dev.com",
        "name": "contact",
        "id": "115341805336974604065",
        "email": "contact@dejesus-gw-rule-dev.com"
      }
    },
    "tags": [
      "forwarded",
      "google-workspace-admin"
    ],
    "cloud": {
      "availability_zone": "us-east1-b",
      "instance": {
        "name": "ubuntu-server-tdejesus",
        "id": "1709224677170316971"
      },
      "provider": "gcp",
      "service": {
        "name": "GCE"
      },
      "machine": {
        "type": "e2-medium"
      },
      "project": {
        "id": "elastic-security-dev"
      },
      "account": {
        "id": "elastic-security-dev"
      }
    },
    "input": {
      "type": "httpjson"
    },
    "@timestamp": "2022-09-13T17:35:49.843Z",
    "ecs": {
      "version": "8.2.0"
    },
    "related": {
      "ip": [
        "2603:6010:ed07:5cb7:73:3153:929f:34bf"
      ],
      "user": [
        "contact"
      ]
    },
    "google_workspace": {
      "actor": {
        "type": "USER"
      },
      "kind": "admin#reports#activity",
      "admin": {
        "user_defined_setting": {
          "name": "99c62"
        },
        "setting": {
          "name": "MESSAGE_SECURITY_RULE",
          "description": "Malicious Email Route"
        }
      },
      "event": {
        "type": "EMAIL_SETTINGS"
      }
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "google_workspace.admin"
    },
    "organization": {
      "id": "C02aasz56"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-09-13T17:40:20Z",
      "provider": "admin",
      "created": "2022-09-13T17:40:19.345Z",
      "action": "CREATE_GMAIL_SETTING",
      "id": "-8920902594508471489",
      "category": [
        "iam"
      ],
      "type": [
        "change",
        "creation"
      ],
      "dataset": "google_workspace.admin"
    },
    "user": {
      "domain": "dejesus-gw-rule-dev.com",
      "name": "contact",
      "id": "115341805336974604065"
    }
  },
  "fields": {
    "elastic_agent.version": [
      "8.4.0"
    ],
    "event.category": [
      "iam"
    ],
    "source.user.email": [
      "contact@dejesus-gw-rule-dev.com"
    ],
    "cloud.availability_zone": [
      "us-east1-b"
    ],
    "source.user.name.text": [
      "contact"
    ],
    "source.geo.region_name": [
      "Ohio"
    ],
    "google_workspace.actor.type": [
      "USER"
    ],
    "source.ip": [
      "2603:6010:ed07:5cb7:73:3153:929f:34bf"
    ],
    "agent.name": [
      "ubuntu-server-tdejesus"
    ],
    "google_workspace.admin.user_defined_setting.name": [
      "99c62"
    ],
    "source.geo.region_iso_code": [
      "US-OH"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "google_workspace.event.type": [
      "EMAIL_SETTINGS"
    ],
    "source.geo.city_name": [
      "North Canton"
    ],
    "user.id": [
      "115341805336974604065"
    ],
    "input.type": [
      "httpjson"
    ],
    "data_stream.type": [
      "logs"
    ],
    "tags": [
      "forwarded",
      "google-workspace-admin"
    ],
    "related.user": [
      "contact"
    ],
    "google_workspace.admin.setting.description": [
      "Malicious Email Route"
    ],
    "event.provider": [
      "admin"
    ],
    "cloud.provider": [
      "gcp"
    ],
    "cloud.machine.type": [
      "e2-medium"
    ],
    "agent.id": [
      "f4011165-a1b8-4d8e-b902-56ba8835cc28"
    ],
    "cloud.service.name": [
      "GCE"
    ],
    "ecs.version": [
      "8.2.0"
    ],
    "event.created": [
      "2022-09-13T17:40:19.345Z"
    ],
    "organization.id": [
      "C02aasz56"
    ],
    "agent.version": [
      "8.4.0"
    ],
    "source.user.name": [
      "contact"
    ],
    "source.as.number": [
      10796
    ],
    "user.name": [
      "contact"
    ],
    "source.geo.location": [
      {
        "coordinates": [
          -81.3798,
          40.7961
        ],
        "type": "Point"
      }
    ],
    "cloud.instance.id": [
      "1709224677170316971"
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "google_workspace"
    ],
    "related.ip": [
      "2603:6010:ed07:5cb7:73:3153:929f:34bf"
    ],
    "source.geo.country_iso_code": [
      "US"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "source.user.id": [
      "115341805336974604065"
    ],
    "user.domain": [
      "dejesus-gw-rule-dev.com"
    ],
    "source.as.organization.name.text": [
      "TWC-10796-MIDWEST"
    ],
    "elastic_agent.id": [
      "f4011165-a1b8-4d8e-b902-56ba8835cc28"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "source.geo.continent_name": [
      "North America"
    ],
    "source.as.organization.name": [
      "TWC-10796-MIDWEST"
    ],
    "google_workspace.admin.setting.name": [
      "MESSAGE_SECURITY_RULE"
    ],
    "google_workspace.kind": [
      "admin#reports#activity"
    ],
    "event.action": [
      "CREATE_GMAIL_SETTING"
    ],
    "event.ingested": [
      "2022-09-13T17:40:20.000Z"
    ],
    "@timestamp": [
      "2022-09-13T17:35:49.843Z"
    ],
    "cloud.account.id": [
      "elastic-security-dev"
    ],
    "event.type": [
      "change",
      "creation"
    ],
    "data_stream.dataset": [
      "google_workspace.admin"
    ],
    "source.user.domain": [
      "dejesus-gw-rule-dev.com"
    ],
    "agent.ephemeral_id": [
      "328620d6-3e81-4b39-863a-bf064564dfa0"
    ],
    "source.geo.country_name": [
      "United States"
    ],
    "event.id": [
      "-8920902594508471489"
    ],
    "event.dataset": [
      "google_workspace.admin"
    ],
    "cloud.instance.name": [
      "ubuntu-server-tdejesus"
    ],
    "cloud.project.id": [
      "elastic-security-dev"
    ],
    "user.name.text": [
      "contact"
    ]
  }
}
```

</details>

### CODE REVIEW

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/integrations/google_workspace/collection_google_workspace_custom_gmail_route_created_or_modified.toml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+### Important Information Regarding Google Workspace Event Lag Times
+- As per Google's documentation, Google Workspace administrators may observe lag times ranging from minutes up to 3 days between the time of an event's occurrence and the event being visible in the Google Workspace admin/audit logs.
+- This rule is configured to run every 10 minutes with a lookback time of 130 minutes.
+- To reduce the risk of false negatives, consider reducing the interval that the Google Workspace (formerly G Suite) Filebeat module polls Google's reporting API for new events.
+- By default, `var.interval` is set to 2 hours (2h). Consider changing this interval to a lower value, such as 10 minutes (10m).
+- See the following references for further information:
+  - https://support.google.com/a/answer/7061566
+  - https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-google_workspace.html"""
+references = ["https://support.google.com/a/answer/1247799?hl=en"]
```
Comment (terrancedejesus) [inline rules/integrations/google_workspace/collection_google_workspace_custom_gmail_route_created_or_modified.toml:42]:
```suggestion
references = ["https://support.google.com/a/answer/2685650?hl=en"]
```

File: rules/integrations/google_workspace/collection_google_workspace_custom_gmail_route_created_or_modified.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+"""
+false_positives = [
+    """
+    Administrators may create custom email routes in Google Workspace based on organizational policies, administrative
+    preference or for security purposes regarding spam.
+    """,
+]
+from = "now-130m"
+index = ["filebeat-*", "logs-google_workspace*"]
+interval = "10m"
```
Comment (brokensound77) [inline rules/integrations/google_workspace/collection_google_workspace_custom_gmail_route_created_or_modified.toml:26]: This seems like excessive overlap to look back 130m every 10m. If the lookback must be so wide, can you update the interval to run every 120m or similar

File: rules/integrations/google_workspace/collection_google_workspace_custom_gmail_route_created_or_modified.toml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+updated_date = "2022/09/13"
+
+[rule]
+author = ["Elastic"]
+description = """
+Gmail is by default the mail routing service in Google Workspace. By default, all email from current Google Workspace
+users with accounts are routed through a domain's mail server for inbound and outbound mail. Adversaries can add a
+custom e-mail route for outbound mail to route these e-mails to their own inbox of choice for data gathering. This
+allows adversaries to capture sensitive information from e-mail and potential attachments, such as invoices or payment
+documents.
```
Comment (brokensound77) [inline rules/integrations/google_workspace/collection_google_workspace_custom_gmail_route_created_or_modified.toml:16]:
```suggestion
Gmail is the default mail routing service in Google Workspace. By default, all email from current Google Workspace
users with accounts are routed through a domain's mail server for inbound and outbound mail. Adversaries can add a
custom e-mail route for outbound mail to route these e-mails to their own inbox of choice for data gathering. This
allows adversaries to capture sensitive information from e-mail and potential attachments, such as invoices or payment
documents.
```

Can you reorder this to put the concise threat up front and the contextual description after. It is great context to have for the description, but having it up front helps with managing multiple rules and alerts

**Review by brokensound77 — APPROVED** [review]
a few comments then LGTM 👍

File: rules/integrations/google_workspace/collection_google_workspace_custom_gmail_route_created_or_modified.toml (hunk 3 above)
Comment (terrancedejesus) [reply rules/integrations/google_workspace/collection_google_workspace_custom_gmail_route_created_or_modified.toml:16]: Roger that

**Pull request merged**